src/js/components/ShareButton.jsx

```diff
@@ -9,19 +17,42 @@ class ShareButton extends React.Component {
 			showing: false
 		}
 	}
-
+	
 	render () {
+
+		// Generate ShareServlet sharing url
+		let metaProps = this.props.meta;
+		if (!metaProps)
+			metaProps = this.props;
+		let params = [];
+		if (metaProps.title)
+			params.push('title=' + escape(metaProps.title));
+		if (metaProps.image)
+			params.push('image=' + escape(metaProps.image));
+		if (metaProps.description)
+			params.push('desc=' + escape(metaProps.description));
+		let url = "https://as.good-loop.com/share?"; //title=Foo&link=" + escape(this.props.url);
+		let paramCount = 0;
+		params.forEach (param => {
+			url += (paramCount != 0 ? "&" : "") + param;
+			paramCount ++;
+		});
+		url += (paramCount != 0 ? "&" : "") + "link=" + escape(this.props.url);
+		console.log("Sharing: " + url);
+		url = escape(url);
+		console.log("Escaped: " + url);
+
```

Comment:
Hm - Let's make a util function in miscutils.ts for url building. Or does modern js have one built in? @roscoemcinerney?

Reply:
https://stackoverflow.com/questions/486896/adding-a-parameter-to-the-url-with-javascript/44160941#44160941

```
let url = new URL("http://foo.bar/");
url.searchParams.set('x', 42);
url.toString();
```

Reply:
That's not supported in IE, so best practice would be to write a utility function in miscutils.ts. That way we can put in an IE shim if we need one (without having to dig up every usage of searchParams).

Also the utility function could take an object of key-values and set them all.

Reply:
Actually: I just tested URLSearchParams. It encodes space as `+`, which should be considered unsafe (not all decoding functions will decode that).

So: best AVOID searchParams. And let's definitely write our own utility method.

Reply:
Done a quick google and found JS does have URL building utils. I'll swap them out